siliconcompiler/core.py

```diff
@@ -4269,66 +4252,78 @@ def show(self, filename=None, extra_options=None):
             self.logger.error('Try passing in a full path to show() instead.')
             return False
 
-        self.logger.info('Showing file %s', filename)
+        self.logger.info(f'Showing file {filename}')
 
-        # Parsing filepath
         filepath = os.path.abspath(filename)
-        filetype = utils.get_file_ext(filepath)
-        localfile = os.path.basename(filepath)
-        if localfile.endswith('.gz'):
-            localfile = os.path.splitext(localfile)[0]
 
         #Check that file exists
         if not os.path.isfile(filepath):
             self.logger.error(f"Invalid filepath {filepath}.")
             return False
 
-        # Opening file from temp directory
-        cwd = os.getcwd()
-        showdir = self.get('option','builddir') + "/_show"
-        os.makedirs(showdir, exist_ok=True)
-        os.chdir(showdir)
-
-        # Uncompress file if necessary
-        if os.path.splitext(filepath)[1].lower() == ".gz":
-            with gzip.open(filepath, 'rb') as f_in:
-                with open(localfile, 'wb') as f_out:
-                    shutil.copyfileobj(f_in, f_out)
-        else:
-            shutil.copy(filepath, localfile)
-
-        #Figure out which tool to use for opening data
-        if filetype in self.getkeys('option','showtool'):
-            # Using env variable and manifest to pass arguments
-            os.environ['SC_FILENAME'] = localfile
-            # Setting up tool
-            tool = self.get('option','showtool', filetype)
-            step = 'show'+filetype
-            index = "0"
-            self.set('arg', 'step', step)
-            self.set('arg', 'index', index)
-            setup_tool = self.find_function(tool, 'setup', 'tools')
-            setup_tool(self, mode='show')
-            self.write_manifest("sc_manifest.tcl", abspath=True)
-            self.write_manifest("sc_manifest.json", abspath=True)
-            self.set('arg', 'step', None)
-            self.set('arg', 'index', None)
-
-            exe = self._getexe(tool)
-            if shutil.which(exe) is None:
-                self.logger.error(f'Executable {exe} not found.')
-                success = False
-            else:
-                # Running command
-                cmdlist = self._makecmd(tool, step, index, extra_options=extra_options)
-                proc = subprocess.run(cmdlist)
-                success = proc.returncode == 0
-        else:
+        saved_config = self.schema.copy()
+
+        filetype = utils.get_file_ext(filepath)
+
+        if filetype not in self.getkeys('option', 'showtool'):
             self.logger.error(f"Filetype '{filetype}' not set up in 'showtool' parameter.")
+            return False
+
+        # Setup new show/screenshot flow
+        flow = 'showflow'
```

Comment:
Can this be a flow inside the "flows" directory?

Reply:
Yes, I originally had it that way, but switched it back. I'll make it a loadable flow